src/main/java/org/testng/ITestResult.java

```diff
@@ -101,4 +101,17 @@
      * @param name - The new name to be used as a test name
      */
     void setTestName(String name);
+
+  /**
+   * @return - <code>true</code> if the test was retried again by an implementation of {@link
+   *     IRetryAnalyzer}
+   */
+  default boolean wasRetried() {
```

Comment:
I think it is not possible to change the implementation. If true, internal impl are the only ones possible and we can change the interface without having a default.

Reply:
I didn't quite get you. Can you please elaborate ? I added default methods because I wasn't sure if there are downstream implementations of this interface and I didn't want to break backward compatibility.

Reply:
Yes, but I'm not sure it is possible to have a custom implementation.

Reply:
Fixed this.